future/name-resolution/002-nr-host-synchronization/README.md

```diff
@@ -128,68 +140,78 @@ All this output is a little overwhelming. You can click on each command to see t
+
+Next, we know that `host-c` is configured through `/etc/nsswitch.conf` to use "multicast" to look up any `.local` address.
+
+> But, what is this "multicast" anyway? We haven't really talked about it. Strictly speaking, a multicast IP address is any IP address within the `224.0.0.0/4` subnet. Any time a message is sent to an IP address in this range, it will be "multicasted". For an Ethernet network like the one that our hosts are both on, this really just means that the packet will be broadcast to every machine on that network.
+
+However, beyond just broadcasting to every machine, multicast has one other interesting property, and that is that specific multicast IP addresses are registered for specific uses. In our case, `224.0.0.251` is reserved for multicast DNS requests. So we're seeing, in this first tcpdump entry, a message being sent to all machines on this ethernet network to this special IP address using the port `5353`, which is also reserved for multicast DNS requests.
 
 So, on this line, we see that `host-c` (`6.0.0.103`) is making a request to a known multicast DNS IP address and port for name resolution for `host-f.local`.
 
+Okay, so what is the next packet that we see chronologically in these tcpdumps?
+
 > **host-f:**
 > 19:59:53.972447 IP (tos 0x0, ttl 255, id 50541, offset 0, flags [DF], proto UDP (17), length 58)
     6.0.0.103.5353 > 224.0.0.251.5353: [bad udp cksum 0xe799 -> 0x628b!] 0 A (QM)? host-f.local. (30)
 
 We see the name resolution request packet that `host-c` sent hitting `host-f`. This is the exact same packet we saw above, except this time from `host-f`'s perspective.
 
-Why is `host-f` receiving a packet for `224.0.0.251`? That's not its IP address!
+So why is `host-f` receiving a packet for `224.0.0.251`? Because it's being *broadcast* on this ethernet network.
+
+We've seen something similar in previous chapters: ARP ([a quick reminder on how ARP and ethernet works](../../../appendix/ip-and-mac-addresses.md)). ARP is a protocol that enables IP discovery between machines on a network. ARP doesn't need to know the IP address of each machine ahead of time because it can send packets to all machines to find out which machine owns an address. Similarly, multicast doesn't need to know the IP addresses of the hosts it wants to communicate with. Both ARP and multicast using the same underlying capability within the ethernet, namely, ethernet broadcast.
 
-So now, let's go back to the question of "what is this 'multicast' stuff?" Multicast is a protocol that provides a structure for a machine on an internetwork to communicate with other machines on an internetwork without already knowing their IP addresses.
+<!-- consider deleting this next paragraph -->
```

Comment:
I think you're correct that this doesn't belong in this README. However... I would advocate that it's interesting (and maybe even useful) information. How would you feel about working it into the [ip-and-mac-addresses](https://github.com/psbanka/build-your-own-internet/blob/main/appendix/ip-and-mac-addresses.md) document instead?